src/alda/parser.clj

```diff
@@ -20,16 +20,12 @@
           :slur              (constantly :slur)
           :flat              (constantly :flat)
           :sharp             (constantly :sharp)
+          :natural           (constantly :natural)
           :dots              #(hash-map :dots (count %))
           :note-length       #(list* 'alda.lisp/note-length %&)
           :duration          #(list* 'alda.lisp/duration %&)
-          :pitch             (fn [s]
-                               (list* 'alda.lisp/pitch
-                                      (keyword (str (first s)))
-                                      (map #(case %
-                                              \- :flat
-                                              \+ :sharp)
-                                           (rest s))))
+          :pitch             #(list* 'alda.lisp/pitch %&)
+          :pitch-notation    #(list* 'alda.lisp/parse-pitch %)
```

Comment:
The parsing library we're using, Instaparse, allows us to [hide tags](https://github.com/Engelberg/instaparse#hiding-tags), which is what I'm proposing here -- each node that needs to capture a `pitch-notation` node can still do so -- the content of the `pitch-notation` node will just be pulled up into the next highest level.

e.g. "c++" will parse as `(pitch :c :sharp :sharp)` instead of `(pitch (pitch-notation "c" "+" "+")`

We would doing the pitch-notation parsing earlier on, and making it more of a implementation detail as part of the parsing process.